src/sqs/index.ts

```diff
@@ -346,13 +352,18 @@ export class SqsQueue<MessagePayload> {
 			await this.deleteMessage(message, log)
 		} catch (err) {
 
-			statsd.increment(sqsQueueMetrics.failed, this.metricsTags)
-			log.error({err}, "Error while executing SQS message")
-
 			try {
 				const errorHandlingResult = await this.errorHandler(err, context);
-				if (!errorHandlingResult.retryable) {
-					log.info("Deleting the message because it is not retryable")
+
+				if(errorHandlingResult.isFailure) {
+					log.error({err}, "Error while executing SQS message")
+					statsd.increment(sqsQueueMetrics.failed, this.metricsTags)
+				} else {
+					log.info({err}, "Expected exception while executing SQS message. Not an error, deleting the message.")
+				}
+
+				if (!errorHandlingResult.retryable || !errorHandlingResult.isFailure) {
```

Comment:
It would be nice if all these conditions would be encapsulated in functions with well-readable names. Something like `isRetryableError()`, `hasReachedMaximumRetries()`, and `shouldBeDelayed()`